tests/dags/providers/provider_api_scripts/test_inaturalist.py

```diff
@@ -41,16 +23,125 @@
 # ---> Reading file photos.csv.gz
 # 36 records
 
+INAT = inaturalist.inaturalistDataIngester()
+RECORD0 = {
+    "foreign_id": 191018903,
+    "filetype": "jpg",
+    "license_url": "http://creativecommons.org/licenses/by-nc/4.0/",
+    "width": 818,
+    "height": 741,
+    "foreign_landing_url": "https://www.inaturalist.org/photos/191018903",
+    "image_url": "https://inaturalist-open-data.s3.amazonaws.com/photos/191018903/medium.jpg",
+    "creator": "tiwane",
+    "creator_url": "https://www.inaturalist.org/users/28",
+    "title": "Ranunculus occidentalis",
+    "tags": "Tracheophyta; Angiospermae; Magnoliopsida; Ranunculales; Ranunculaceae; Ranunculoideae; Ranunculeae; Ranunculus",
+}
+
+
+def test_get_next_query_params_no_prior():
+    expect = {"page_number": 0}
+    actual = INAT.get_next_query_params()
+    assert expect == actual
+
+
+def test_get_next_query_params_prior_0():
+    expect = {"page_number": 1}
+    actual = INAT.get_next_query_params({"page_number": 0})
+    assert expect == actual
+
+
 @pytest.mark.parametrize(
-    "file_name, expected",
+    "file_name, expect",
     [
-        ('00_create_schema.sql', [('inaturalist', )]),
-        ('01_photos.sql', [(36, )] ),
-        ('02_observations.sql', [(31, )] ),
-        ('03_taxa.sql', [(183, )] ),
-        ('04_observers.sql', [(22, )] ),
-    ]
+        ("00_create_schema.sql", [("inaturalist",)]),
+        ("03_taxa.sql", [(183,)]),
+        ("04_observers.sql", [(22,)]),
+    ],
 )
-def test_load_from_s3(file_name, expected):
-    actual = inaturalist.run_sql_file(file_name)
-    assert actual == expected
+def test_sql_runner(file_name, expect):
+    actual = inaturalist.sql_runner(file_name)
+    assert actual == expect
+
+
+def test_load_photos():
+    expect = [(36,)]
+    actual = inaturalist.load_photos()
+    assert actual == expect
+
+
+def test_load_observations():
+    expect = [(31,)]
+    actual = inaturalist.load_observations()
+    assert actual == expect
+
+
+def test_get_response_json():
+    expect0 = RECORD0
+    actual = INAT.get_response_json({"page_number": 0})
+    assert isinstance(actual, list)
+    assert len(actual) == 33
+    assert actual[0][0] == expect0
+
+
+def test_get_batch_data_none_response():
+    expect = None
+    actual = INAT.get_batch_data(None)
+    assert actual == expect
+
+
+def test_get_batch_data_empty_response():
+    expect = None
+    actual = INAT.get_batch_data({})
+    assert actual == expect
```

Comment:
Here's a quick way to combine these with parameterization, if you're interested!
```suggestion
@pytest.mark.parametrize("value", [None, {}])
def test_get_batch_data_returns_none(value):
    actual = INAT.get_batch_data(value)
    assert actual == None
```

Reply:
Awesome! [Thanks](https://github.com/WordPress/openverse-catalog/blob/7b0854a0fe4adc26979b175f96540847d410cae3/tests/dags/providers/provider_api_scripts/test_inaturalist.py#L52)!